rskj-core/src/main/java/co/rsk/core/DifficultyCalculator.java

```diff
@@ -103,4 +148,38 @@ private static BlockDifficulty calcDifficultyWithTimeStamps(
         // Note that we have to apply max() first in case fromParent ended up being negative.
         return new BlockDifficulty(max(minDif.asBigInteger(), fromParent));
     }
+
+    private static BlockDifficulty calcSuperDifficultyWithTimeStamps(
+            BlockDifficulty oldDifficulty,
+            BigInteger correctionFactor,
+            BigInteger superBlockDifficultyCorrectionFactor,
+            BigInteger superBlockDifficultyMinCorrectionFactor,
+            BigInteger superBlockDifficultyMaxCorrectionFactor) {
+
+        if(correctionFactor.compareTo(superBlockDifficultyMinCorrectionFactor) < 0) {
+            var newSuperDifficulty = oldDifficulty
+                    .multiply(superBlockDifficultyMinCorrectionFactor)
+                    .asBigInteger()
+                    .divide(superBlockDifficultyCorrectionFactor);
+
+            return new BlockDifficulty(newSuperDifficulty);
+        }
+
+        if(correctionFactor.compareTo(superBlockDifficultyMaxCorrectionFactor) > 0) {
+            var newSuperDifficulty = oldDifficulty
+                    .multiply(superBlockDifficultyMaxCorrectionFactor)
+                    .asBigInteger()
+                    .divide(superBlockDifficultyCorrectionFactor);
+
+            return new BlockDifficulty(newSuperDifficulty);
+        }
+
+        var newSuperDifficulty = oldDifficulty
```

Comment:
maybe we could extract this operations to a new function that returns BlockDifficulty, and receives the correction factors, something like this:

```java
private BlockDifficulty getSuperDifficulty(oldDifficulty, correctionFactor, difficultyCorrectionFactor) {
    var newSuperDifficulty = oldDifficulty
                    .multiply(correctionFactor)
                    .asBigInteger()
                    .divide(difficultyCorrectionFactor);

     return new BlockDifficulty(newSuperDifficulty);
}
```

And then call that function inside the if blocks. WDYT of that?

Reply:
Addressed